classes/publication/DAO.php

```diff
@@ -489,25 +490,27 @@ protected function deleteControlledVocab(int $publicationId)
      */
     protected function setCategories(Publication $publication)
     {
-        $publication->setData(
-            'categoryIds',
-            Repo::category()->getCollector()
-                ->filterByPublicationIds([$publication->getId()])
-                ->getIds()
-                ->toArray()
-        );
+        try {
```

Comment:
I'd rather leave the exception to bubble upwards if necessary, rather than wrapping it in a `try` / `catch` / `error_log`. (Also below.) As it stands, this prevents calling code from being able to catch errors.